index.js

```diff
@@ -872,25 +872,27 @@ class ServerlessAppsyncPlugin {
     return `GraphQlDs${this.getCfnName(name)}`;
   }
 
-  processTemplate(template, config) {
+  processTemplate(template, config, tplSubstitutions) {
     // TODO use serverless variable parser and serverless variable syntax config
     const variableSyntax = RegExp(/\${([\w\d-_]+)}/g);
-    const configVariables = Object.keys(config.substitutions);
+    const tplConfigVariables = tplSubstitutions != null ? Object.keys(tplSubstitutions) : [];
+    const configVariables = Object.keys(config.substitutions).concat(tplConfigVariables);
     const templateVariables = [];
     let searchResult;
     // eslint-disable-next-line no-cond-assign
     while ((searchResult = variableSyntax.exec(template)) !== null) {
       templateVariables.push(searchResult[1]);
     }
 
-    const substitutions = configVariables
+    const subs = configVariables
       .filter(value => templateVariables.indexOf(value) > -1)
       .filter((value, index, array) => array.indexOf(value) === index)
       .reduce(
         (accum, value) => Object.assign(accum, { [value]: config.substitutions[value] }),
         {},
       );
 
+    const substitutions = Object.assign({}, subs, tplSubstitutions);
```

Comment:
@deankostomaj On line 879, you assign `configVariables` with a concat of `config.substitutions` and `tplConfigVariables `.
It then goes through the filters on line 887.
But in `reduce` function, you only assign values from `config.substitutions`.
These are the values that get into `subs`.
But then again you concat `subs` with `tplSubstitutions`.

This looks a bit weird. 

I think we can improve this by doing the following: 
- create a new object (e.g.: `allSubstitutions`) with that concats `config.substitutions` and `tplSubstitutions`
- assign `configVariables` with the keys of `allSubstitutions`.
- on line 890, in the `reduce` function, change `config.substitutions[value]` with `allSubstitutions[value]`.

The rest should not need to change.

Reply:
hey. I will fix it in the second half of December when I will have a bit more time. Sorry for the wait.